src/Spryker/Zed/AppPayment/Business/Payment/Method/PaymentMethod.php

```diff
@@ -0,0 +1,213 @@
+    {
+        foreach ($persistedPaymentMethodTransfers as $persistedPaymentMethodTransfer) {
+            if ($paymentMethodName === $persistedPaymentMethodTransfer->getName()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    protected function getPaymentMethodsToDelete(array $persistedPaymentMethodTransfers, array $configuredPaymentMethodTransfers): array
+    {
+        $methodsToDelete = [];
+
+        foreach ($persistedPaymentMethodTransfers as $persistedPaymentMethodTransfer) {
+            if (!$this->isPaymentMethodConfigured($persistedPaymentMethodTransfer->getName(), $configuredPaymentMethodTransfers)) {
+                $methodsToDelete[] = $persistedPaymentMethodTransfer;
+            }
+        }
+
+        return $methodsToDelete;
+    }
+
+    /**
+     * @param array<\Generated\Shared\Transfer\PaymentMethodTransfer> $configuredPaymentMethodTransfers
+     */
+    protected function isPaymentMethodConfigured(string $paymentMethodName, array $configuredPaymentMethodTransfers): bool
+    {
+        foreach ($configuredPaymentMethodTransfers as $configuredPaymentMethodTransfer) {
+            if ($paymentMethodName === $configuredPaymentMethodTransfer->getName()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public function deletePaymentMethods(AppConfigTransfer $appConfigTransfer): AppConfigTransfer
+    {
+        if (!($this->appPaymentPlatformPlugin instanceof AppPaymentPaymentMethodsPlatformPluginInterface)) {
+            $deletePaymentMethodTransfer = new DeletePaymentMethodTransfer();
+            $deletePaymentMethodTransfer
+                ->setName($this->appPaymentConfig->getPaymentMethodName())
+                ->setProviderName($this->appPaymentConfig->getPaymentProviderName());
+
+            $this->paymentMethodMessageSender->sendDeletePaymentMethodMessage($deletePaymentMethodTransfer, $appConfigTransfer);
+
+            return $appConfigTransfer;
+        }
+
+        $tenantIdentifier = $appConfigTransfer->getTenantIdentifierOrFail();
+
+        $paymentMethodConfigurationRequestTransfer = new PaymentMethodConfigurationRequestTransfer();
+        $paymentMethodConfigurationRequestTransfer->setAppConfig($appConfigTransfer);
+
+        $paymentMethodCollectionResponseTransfer = $this->appPaymentPlatformPlugin->configurePaymentMethods($paymentMethodConfigurationRequestTransfer);
+
+        // Get current persisted Tenants payment methods
+        $persistedPaymentMethodTransfers = $this->appPaymentRepository->getTenantPaymentMethods($tenantIdentifier);
+
+        // Get current configured payment methods
+        $configuredPaymentMethodTransfers = $paymentMethodCollectionResponseTransfer->getPaymentMethods()->getArrayCopy();
+
+        $paymentMethodTransfersToDelete = $this->getPaymentMethodsToDelete($persistedPaymentMethodTransfers, $configuredPaymentMethodTransfers);
```

Comment:
Update this, all persisted payment methods can be deleted. No need to ask the Platform plugin.